More marine DA j-jobs

guillaumevernieres opened the pull request:
**Description**
Addition of 2 j-jobs:
- jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT_VRFY
This job currently dumps a few files of the impulse response of B. Not meant to ever go in ops.
- jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY
This job will eventually generate relevant figures from the marine DA cycle. It currently points to a script that does not yet exist.


I also took this PR opportunity to:
- update how we load the modules, following what was done for the atmosphere.
- modify ```ush/load_ufsda_modules.sh``` so we could optionally specify what to load (GDAS or EVA), default is GDAS when no argument is present 
- Use POST to copy what's needed in `COM`

- fixes #1273 

**Type of change**

- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
GDASApp CI/Container
GDASApp CI/Orion for the `JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT_VRFY` jjob

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

guillaumevernieres committed:
bmat_vrfy and ana_vrfy jobs

guillaumevernieres committed:
forgotten files ...

guillaumevernieres committed:
tidy run job

guillaumevernieres committed:
code norm

guillaumevernieres commented:
preemptively assuming the last commit fixes the linter check ...

aerorahul commented on jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT_VRFY:
```diff
@@ -0,0 +1,45 @@
+source "${HOMEgfs}/ush/preamble.sh"
+
+
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
```
Is there a reason why every `DATA` is `${RUN}ocnanl_${cyc}`?
We need this solely if it will need to be preserved after this job is completed.
If it is not needed, then we should stick with the default `${job}.${jobid}` that are defined upstream.

This comment applies to the `JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY` as well.

guillaumevernieres replied:
yes, at least for now it needs to be preserved.

aerorahul replied:
This directory is being gutted here:
https://github.com/NOAA-EMC/global-workflow/blob/f98433ff3661ef848d381ae65322623573a114cc/jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST#L21

I suggest the post job to copy what it needs to `COM` and create a working directory for this job in `$DATAROOT`

guillaumevernieres replied:
ok, will look into it @aerorahul

guillaumevernieres replied:
@aerorahul , I created an issue [here](https://github.com/NOAA-EMC/global-workflow/issues/1273) to cleanup how we keep what we need from the cycle. Happy to add the necessary changes to that PR, or to do it in a different PR.

WalterKolczynski-NOAA replied:
To elaborate a bit further: `$DATA` should only contain temporary data during runtime and is wiped at the end of the job. Typically that means a random directory name is okay. With the prep/run/post paradigm, that doesn't work because RUN needs to know where PREP staged stuff, but prep/run/post should still be considered a single job for purposes of `$DATA` and deleted at the end of POST. Anything that is needed outside of that "one" job needs to get placed into `$COM`.

guillaumevernieres replied:
Understood @WalterKolczynski-NOAA , I'll push the modifications that address that issue (#1273 ) later today ... maybe. 
I'll use the default `$DATA` (default is `${job}.${jobid}`?) for the `JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY`, that job will use data saved in `$COM`. The rest of the marine DA j-jobs all depend on PREP, so I'll have to stick to the specified `$DATA=${RUN}ocnanal_${cyc}`.

aerorahul commented on jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY:
```diff
@@ -0,0 +1,45 @@
+
+
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalrun" -c "base ocnanal ocnanalprep"
```
Is this job going to use the same resources of `ocnanalrun`? I highly doubt that.

guillaumevernieres replied:
oops, nope.

aerorahul reviewed (commented):
Most changes look ok.
I have a few clarifying questions that probably @CoryMartin-NOAA could answer better as he knows how the `env` files are used to set variables for jobs.

guillaumevernieres committed:
wrong resources

guillaumevernieres commented:
humm ... looks like I pushed the shellcheck fix to the aether!

guillaumevernieres committed:
more shellcheck

AndrewEichmann-NOAA reviewed (commented):
Looks plausible

RussTreadon-NOAA commented on jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT_VRFY:
```diff
@@ -0,0 +1,45 @@
+###############################################################
+# Run relevant script
+
+EXSCRIPT=${GDASPREPPY:-${HOMEgfs}/sorc/gdas.cd/scripts/exgdas_global_marine_analysis_bmat_vrfy.sh}
```
Will this script eventually move to g-w `scripts`?

guillaumevernieres replied:
Yes, at some point in the future @RussTreadon-NOAA .

RussTreadon-NOAA replied:
OK.  I understand the utility in keeping it in GDASApp for the time being.

RussTreadon-NOAA commented on jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY:
```diff
@@ -0,0 +1,45 @@
+###############################################################
+# Run relevant script
+
+EXSCRIPT=${GDASPREPPY:-${HOMEgfs}/sorc/gdas.cd/scripts/exgdas_global_marine_analysis_vrfy.sh}
```
Same question as before.   Will this script eventually move to g-w `scripts/`?

RussTreadon-NOAA replied:
Got it.

RussTreadon-NOAA reviewed (approved):
Changes look straightforward to me.    Approve.

guillaumevernieres committed:
cp stuff to com

guillaumevernieres committed:
fixed shell norms

guillaumevernieres commented:
@aerorahul , @WalterKolczynski-NOAA , I think I addressed your comments. 
It "works" with a companion branch that I have yet to push in the GDASApp,
`POST` is now actually doing something, I'll move the file manipulation in a dedicated script later, as @aerorahul suggested.
```console
gvernier:gw$ ctest 
Test project /home/gvernier/wrk/gw-nameupdate/sorc/gdas.cd/build/test/soca/gw
    Start 1: test_gdasapp_soca_run_clean
1/5 Test #1: test_gdasapp_soca_run_clean ..........................   Passed    0.01 sec
    Start 2: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP
2/5 Test #2: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP ...   Passed    4.16 sec
    Start 3: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN
3/5 Test #3: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN ....   Passed   23.65 sec
    Start 4: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_POST
4/5 Test #4: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_POST ...   Passed    1.52 sec
    Start 5: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY
5/5 Test #5: test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY ...   Passed    3.91 sec

100% tests passed, 0 tests failed out of 5

Total Test time (real) =  33.26 sec
```

guillaumevernieres commented:
@WalterKolczynski-NOAA , @aerorahul , I've addressed your comments. Could I get a quick review/merge?
